From 6d2f8265c58974b31a10ef060e72c30233a19f39 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ga=C3=ABl=20Bonithon?= <gael@xfce.org>
Date: Fri, 7 Jun 2024 12:32:41 +0200
Subject: [PATCH] systray-icon: Make setting a boolean

This was already the case in practice, since the binding in the settings
dialog stored this setting as a boolean in xfconf, and the values
defined in XfpmShowIcon were not used anywhere, except for the default
value in XfpmManager. This inconsistency ended up causing a regression
following 09a9f417, as it previously only worked by chance.

Fixes: #234
---
 common/xfpm-config.h    |  2 +-
 common/xfpm-enum-glib.h |  9 ---------
 src/xfpm-manager.c      | 18 +++++++++---------
 src/xfpm-xfconf.c       | 10 ++++------
 4 files changed, 14 insertions(+), 25 deletions(-)

diff --git a/common/xfpm-config.h b/common/xfpm-config.h
index 7598e981..172fdd1f 100644
--- a/common/xfpm-config.h
+++ b/common/xfpm-config.h
@@ -73,7 +73,7 @@ G_BEGIN_DECLS
 #define HEARTBEAT_COMMAND "heartbeat-command"
 #define DEFAULT_HEARTBEAT_COMMAND NULL
 #define SHOW_TRAY_ICON "show-tray-icon"
-#define DEFAULT_SHOW_TRAY_ICON SHOW_ICON_WHEN_BATTERY_PRESENT
+#define DEFAULT_SHOW_TRAY_ICON FALSE
 
 #define POWER_BUTTON_ACTION "power-button-action"
 #define DEFAULT_POWER_BUTTON_ACTION XFPM_DO_NOTHING
diff --git a/common/xfpm-enum-glib.h b/common/xfpm-enum-glib.h
index 51ba8d23..0cea62c1 100644
--- a/common/xfpm-enum-glib.h
+++ b/common/xfpm-enum-glib.h
@@ -85,15 +85,6 @@ typedef enum
   N_SPIN_DOWN_HDDS
 } XfpmSpindownRequest;
 
-typedef enum
-{
-  SHOW_ICON_ALWAYS,
-  SHOW_ICON_WHEN_BATTERY_PRESENT,
-  SHOW_ICON_WHEN_BATTERY_CHARGING_DISCHARGING,
-  SHOW_ICON_NEVER,
-  N_SHOW_ICONS
-} XfpmShowIcon;
-
 typedef enum
 {
   PANEL_LABEL_NONE,
diff --git a/src/xfpm-manager.c b/src/xfpm-manager.c
index 6349afe7..2675c79f 100644
--- a/src/xfpm-manager.c
+++ b/src/xfpm-manager.c
@@ -107,7 +107,7 @@ struct XfpmManagerPrivate
   XfpmIdle *idle;
   GtkStatusIcon *adapter_icon;
   GtkWidget *power_button;
-  gint show_tray_icon;
+  gboolean show_tray_icon;
   XfpmDpms *dpms;
   GTimer *timer;
   gint inhibit_fd;
@@ -136,11 +136,11 @@ xfpm_manager_class_init (XfpmManagerClass *klass)
                           | G_PARAM_STATIC_BLURB)
 
   g_object_class_install_property (object_class, PROP_SHOW_TRAY_ICON,
-                                   g_param_spec_int (SHOW_TRAY_ICON,
-                                                     SHOW_TRAY_ICON,
-                                                     SHOW_TRAY_ICON,
-                                                     0, N_SHOW_ICONS - 1, DEFAULT_SHOW_TRAY_ICON,
-                                                     XFPM_PARAM_FLAGS));
+                                   g_param_spec_boolean (SHOW_TRAY_ICON,
+                                                         SHOW_TRAY_ICON,
+                                                         SHOW_TRAY_ICON,
+                                                         DEFAULT_SHOW_TRAY_ICON,
+                                                         XFPM_PARAM_FLAGS));
 #undef XFPM_PARAM_FLAGS
 }
 
@@ -786,11 +786,11 @@ xfpm_manager_set_property (GObject *object,
   switch (property_id)
   {
     case PROP_SHOW_TRAY_ICON:
-      new_value = g_value_get_int (value);
+      new_value = g_value_get_boolean (value);
       if (new_value != manager->priv->show_tray_icon)
       {
         manager->priv->show_tray_icon = new_value;
-        if (new_value > 0)
+        if (new_value)
         {
           if (WINDOWING_IS_X11 ())
             xfpm_manager_show_tray_icon (manager);
@@ -820,7 +820,7 @@ xfpm_manager_get_property (GObject *object,
   switch (property_id)
   {
     case PROP_SHOW_TRAY_ICON:
-      g_value_set_int (value, manager->priv->show_tray_icon);
+      g_value_set_boolean (value, manager->priv->show_tray_icon);
       break;
 
     default:
diff --git a/src/xfpm-xfconf.c b/src/xfpm-xfconf.c
index 716e84b0..5892b252 100644
--- a/src/xfpm-xfconf.c
+++ b/src/xfpm-xfconf.c
@@ -280,12 +280,10 @@ xfpm_xfconf_class_init (XfpmXfconfClass *klass)
    **/
   g_object_class_install_property (object_class,
                                    PROP_TRAY_ICON,
-                                   g_param_spec_uint (SHOW_TRAY_ICON,
-                                                      NULL, NULL,
-                                                      0,
-                                                      N_SHOW_ICONS - 1,
-                                                      DEFAULT_SHOW_TRAY_ICON,
-                                                      G_PARAM_READWRITE));
+                                   g_param_spec_boolean (SHOW_TRAY_ICON,
+                                                         NULL, NULL,
+                                                         DEFAULT_SHOW_TRAY_ICON,
+                                                         G_PARAM_READWRITE));
 
   /**
    * XfpmXfconf::critical-battery-action
